src/nupic/algorithms/Connections.cpp

```diff
@@ -438,45 +453,39 @@ void Connections::adaptSegment(const Segment segment,
 {
   const auto &inputArray = inputs.getDense();
 
-  for( const auto &synapse : synapsesForSegment(segment) ) {
-    const SynapseData &synapseData = dataForSynapse(synapse);
-
-    Permanence permanence = synapseData.permanence;
-    if( inputArray[synapseData.presynapticCell] ) {
-      permanence += increment;
-    } else {
-      permanence -= decrement;
-    }
-
-    updateSynapsePermanence(synapse, permanence);
-  }
-}
+  if( timeseries_ ) {
+    previousUpdates_.resize( synapseFlatListLength(), 0.0f );
+    currentUpdates_.resize(  synapseFlatListLength(), 0.0f );
 
-void Connections::adaptSegment(const Segment segment,
-                               const SDR &inputs,
-                               const Permanence increment,
-                               const Permanence decrement,
-                                     vector<Permanence> &previousUpdates,
-                                     vector<Permanence> &currentUpdates)
-{
-  const auto &inputArray = inputs.getDense();
-  previousUpdates.resize( synapseFlatListLength(), 0.0f );
-  currentUpdates.resize(  synapseFlatListLength(), 0.0f );
+    for( const auto synapse : synapsesForSegment(segment) ) {
+      const SynapseData &synapseData = dataForSynapse(synapse);
 
-  for( const auto &synapse : synapsesForSegment(segment) ) {
-    const SynapseData &synapseData = dataForSynapse(synapse);
+      Permanence update;
+      if( inputArray[synapseData.presynapticCell] ) {
+        update = increment;
+      } else {
+        update = -decrement;
+      }
 
-    Permanence update;
-    if( inputArray[synapseData.presynapticCell] ) {
-      update = increment;
-    } else {
-      update = -decrement;
+      if( update != previousUpdates_[synapse] ) {
+        updateSynapsePermanence(synapse, synapseData.permanence + update);
```

Comment:
can the logic (the "if" above) be moved to updateSynapse, so it applies globally, if set in the bool timeseries?

Reply:
No, because `updateSynapsePermanence` receives the new permanence, not the change done to it.

Also it's called by other methods which should ignore the time series logic, such as `raisePermanencesToThreshold` and that method is call by the ColumnPooler.